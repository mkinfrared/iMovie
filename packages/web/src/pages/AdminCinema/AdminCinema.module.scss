.AdminCinema {
  position: relative;

  > h3 {
    margin-bottom: 20px;
  }
}
