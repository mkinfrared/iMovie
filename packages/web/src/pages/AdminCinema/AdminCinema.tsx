import Typography from "@material-ui/core/Typography";
import AddIcon from "@material-ui/icons/Add";
import React, { useCallback, useState } from "react";

import AddCinemaForm from "containers/AddCinemaForm";
import AdminCinemaList from "containers/AdminCinemaList";
import FloatButton from "ui/FloatButton";

import css from "./AdminCinema.module.scss";

const AdminCinema = () => {
  const [formOpen, setFormOpen] = useState(false);

  const openForm = useCallback(() => {
    setFormOpen(true);
  }, []);

  const closeForm = useCallback(() => {
    setFormOpen(false);
  }, []);

  return (
    <>
      <div className={css.AdminCinema}>
        <Typography variant="h3" align="center">
          Cinemas
        </Typography>
        <AdminCinemaList />
        <AddCinemaForm open={formOpen} onClose={closeForm} />
      </div>
      <FloatButton onClick={openForm}>
        <AddIcon />
      </FloatButton>
    </>
  );
};

export default React.memo(AdminCinema);
