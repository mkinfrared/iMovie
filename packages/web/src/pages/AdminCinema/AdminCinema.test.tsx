import { act, findByText, fireEvent, render } from "@testing-library/react";
import React, { ReactElement } from "react";

import AdminCinema from "./AdminCinema";

jest.mock("components/AddCinemaForm", () => (props: any) => (
  <div data-testid="components/AddCinemaForm">
    <code>{JSON.stringify(props)}</code>
  </div>
));

describe("<AdminCinema />", () => {
  let Component: ReactElement;

  beforeEach(() => {
    Component = <AdminCinema />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should open <AddCinemaForm />", async () => {
    const { container } = render(Component);
    const floatButton = await findByText(container, "+");

    await act(async () => {
      await fireEvent.click(floatButton);
    });

    const result = await findByText(container, /\{"open":true\}/gi);

    expect(result).toBeInTheDocument();
  });
});
