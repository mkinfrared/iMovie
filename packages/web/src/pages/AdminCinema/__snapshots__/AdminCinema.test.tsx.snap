// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminCinema /> should match a snapshot 1`] = `
<div>
  <div
    class="AdminCinema"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
    >
      Cinemas
    </h3>
    <div
      data-testid="containers/AdminCinemaList"
    >
      <code>
        {}
      </code>
    </div>
    <div
      data-testid="containers/AddCinemaForm"
    >
      <code>
        {"open":false}
      </code>
    </div>
  </div>
  <div
    class="FloatButton"
  >
    <button
      class="MuiButtonBase-root MuiFab-root button MuiFab-primary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        +
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
